cmd/snap-confine/mount-support-nvidia.c

```diff
@@ -398,6 +407,15 @@ void sc_mount_nvidia_driver(const char *rootfs_dir)
 	if (access(SC_NVIDIA_DRIVER_VERSION_FILE, F_OK) != 0) {
 		return;
 	}
+
+	int res = mkdir(SC_LIB, 0755);
```

Comment:
Do we need a recursive mkdir() here?

Reply:
This is right before `/var/lib/snapd/lib` will see use in sc_nvidia_{biarch,multiarch}. The corresponding calls are responsible for creating their leaf paths.

Reply:
We _might_ (and that's why I was thinking about the use of the code that already handles that) but at the same time I think having a simple mkdir is fine, this will at most, create the leaf directory.

My suggestion would be to look at a follow-up which focuses on using the right mkdir APIs everywhere in C

Reply:
Ideally we'd want to have `sc_nonfatal_mkpath`, that can take the root, eg. `/var/lib/snapd` (at least this part should have been created by packager), the leaf `lib/{gl,gl32,vulkan}`, the owner (root:root), and the mode. The recursive mkdir that's there now seemed like the least invasive option.

Reply:
After discussing this on irc I see that we don't need a recursive call here. The parent is /var/lib/snapd and if that is missing we have not state file and things are seriously messed up. So this is fine and no need for extra complications.